pysmt/solvers/msat.py

```diff
@@ -1082,16 +1083,21 @@ def declare_variable(self, var):
 if hasattr(mathsat, "MSAT_EXIST_ELIM_ALLSMT_FM"):
     class MSatQuantifierEliminator(QuantifierEliminator, IdentityDagWalker):
 
-        LOGICS = [LRA]
+        LOGICS = [LRA, LIA]
 
-        def __init__(self, environment, logic=None, algorithm='fm'):
+        def __init__(self, environment, logic=None, algorithm='lw'):
             """Initialize the Quantifier Eliminator using 'fm' or 'lw'.
 
             fm: Fourier-Motzkin (default)
             lw: Loos-Weisspfenning
             """
             if algorithm not in ['fm', 'lw']:
                 raise PysmtValueError("Algorithm can be either 'fm' or 'lw'")
+
+            if logic is not None and (not logic <= LRA and algorithm != "lw"):
+                raise NotImplementedError("MathSAT quantifier elimination for LIA"\
```

Comment:
I would change this into a "Pysmt value error", since we cannot implement it, even if we wanted.